rvgo/slow/vm.go

```diff
@@ -132,11 +128,11 @@ func Step(calldata []byte, po PreimageOracle) (stateHash common.Hash, outErr err
 	}
 
 	proofContentOffset := shortToU64(uint16(stateContentOffset) + paddedStateSize + 32)
+	// TODO: validate abi offset values?
```

Comment:
Yes, please :D 

Cannon does this as well.

Reply:
I'm trying to look at where cannon does this, but can't find it since I"m not familiar with the codebase :( Could you please give me quick pointer on where to look?

Reply:
Yeah for sure! Apologies for the late response. Here are the calldata layout checks cannon does: https://github.com/ethereum-optimism/optimism/blob/85248727fe59498ed62788eca8d9c5849cfa5074/packages/contracts-bedrock/src/cannon/MIPS.sol#L216-L231

Reply:
Thanks! It's fixed in https://github.com/ethereum-optimism/asterisc/pull/90/commits/eb345925184f9f364369d10ed8d028c070121c38